x/poe/contract/contractio.go

```diff
@@ -176,6 +180,55 @@ func (a ContractAdapter) doExecute(ctx sdk.Context, msg interface{}, sender sdk.
 	return sdkerrors.Wrap(err, "execute")
 }
 
+// RawPageableResult is a query response type were the pagination key is the raw last element.
+type RawPageableResult interface {
+	// JSOMFieldName returns the raw json field name that contains the pageable elements
+	JSOMFieldName() string
+}
+
+// PageableResult is a query response where the cursor is a subset of the raw last element.
+type PageableResult interface {
+	PaginationCursor() PaginationCursor
+}
+
+// execute a smart query with the contract that returns multiple elements
+// returns a cursor whenever the result set has more than 1 element
+func (a ContractAdapter) doPageableQuery(ctx sdk.Context, query interface{}, result interface{}) (PaginationCursor, error) {
+	if err := a.addressLookupErr; err != nil {
+		return nil, err
+	}
+	bz, err := json.Marshal(query)
+	if err != nil {
+		return nil, sdkerrors.Wrap(err, "marshal query")
+	}
+	res, err := a.twasmKeeper.QuerySmart(ctx, a.contractAddr, bz)
+	if err != nil {
+		return nil, err
+	}
+
+	if err := json.Unmarshal(res, result); err != nil {
+		return nil, sdkerrors.Wrap(err, "unmarshal result")
+	}
+
+	var cursor PaginationCursor
+	switch p := result.(type) {
+	case PageableResult:
+		cursor = p.PaginationCursor()
+	case RawPageableResult:
```

Comment:
Same here. This is cool, but if unused, why add it now?

Reply:
I appreciate the feedback and you have an 👁️  on not over engineering the query! 
I had started with the "raw" query type as this was the common pagination way as far as I remembered. But this is not correct for this contract type.  I will remove this now.